discovery_engine_core/core/src/engine.rs

```diff
@@ -352,6 +362,70 @@ where
             Ok(())
         }
     }
+
+    /// Perform an active search with the given query parameters.
+    pub async fn active_search(
+        &mut self,
+        query: &str,
+        page: usize,
+        page_size: usize,
+    ) -> Result<Vec<Document>, Error> {
+        if query.trim().is_empty() {
+            return Err(Error::InvalidQuery);
+        }
+        let EndpointConfig {
+            api_key,
+            api_base_url,
+            markets,
+            ..
+        } = &self.config;
+
+        let mut errors = Vec::new();
+        let mut articles = Vec::new();
+        let filter = &Filter::default().add_keyword(query);
+        let client = Client::new(api_key.clone(), api_base_url.clone());
+
+        let markets = markets.read().await;
+        let scaled_page_size = page_size / markets.len() + 1;
+        let scaled_page = Some(page * markets.len());
```

Comment:
A page as always `page_size` size so we don't have to change anything for the page.